### PR TITLE
Add `rekordcrate_derive` crate that provides procedural macros

### PR DESCRIPTION
**Pull request by Holzhaus**
Replaces hundreds of lines of manual `parse` method implementations for values in `*SETTING.DAT` files with a simple `#[derive(Parse)]`. Unsure if the additional complexity of using a dedicated derive crate and the additional dependencies is worth it.

@Swiftb0y @uklotzde Feedback would be very much appreciated.

**TODO:**
- [ ] Investigate values of the `PadButtonBrightness` values. Something looks odd there.
- [ ] Check if we should use an `impl trait` instead of a direct `impl`
- [ ] Check if we should use `Deserialize` instead of `Parse` (consistent with serde)

### CODE REVIEW

File: rekordcrate/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "rekordcrate"
+version = "0.1.1"
+authors = ["Jan Holthuis <jan.holthuis@ruhr-uni-bochum.de>"]
+description = "Library for parsing Pioneer Rekordbox device exports"
+readme = "README.md"
```
Comment (Holzhaus) [inline rekordcrate/Cargo.toml:6]: Note to myself: I think this path is wrong now.

File: rekordcrate/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+repository = "https://github.com/Holzhaus/rekordcrate"
+license = "MPL-2.0"
+keywords = ["rekordbox", "dj", "export", "database", "parser"]
+categories = ["parser-implementations"]
+edition = "2021"
+exclude = [".*"]
+
+[dependencies]
+nom = "7"
+rekordcrate_derive = { path = "../rekordcrate_derive" }
```
Comment (Holzhaus) [inline rekordcrate/Cargo.toml:16]: Will this relative path work with `crates.io`? I guess it doesn't...

Comment (uklotzde) [reply rekordcrate/Cargo.toml:16]: https://github.com/serde-rs/serde/blob/master/serde/Cargo.toml

**Comment by Holzhaus** [comment]
Superseeded by #binrw.

**Pull request closed without merging**